persist/sqlite/hosts.go

```diff
@@ -0,0 +1,315 @@
+		if params.V2 {
+			filters = append(filters, "rhp4_prices_contract_price <= ?")
+		} else {
+			filters = append(filters, "settings_contract_price <= ?")
+		}
+		args = append(args, encode(params.MaxContractPrice))
+	}
+	if !params.MaxUploadPrice.IsZero() {
+		if params.V2 {
+			filters = append(filters, "rhp4_prices_ingress_price <= ?")
+		} else {
+			filters = append(filters, "settings_upload_bandwidth_price <= ?")
+		}
+		args = append(args, encode(params.MaxUploadPrice))
+	}
+	if !params.MaxDownloadPrice.IsZero() {
+		if params.V2 {
+			filters = append(filters, "rhp4_prices_egress_price <= ?")
+		} else {
+			filters = append(filters, "settings_download_bandwidth_price <= ?")
+		}
+		args = append(args, encode(params.MaxDownloadPrice))
+	}
+	if !params.MaxBaseRPCPrice.IsZero() {
+		filters = append(filters, "settings_base_rpc_price <= ?")
+		args = append(args, encode(params.MaxBaseRPCPrice))
+	}
+	if !params.MaxSectorAccessPrice.IsZero() {
+		filters = append(filters, "settings_sector_access_price <= ?")
+		args = append(args, encode(params.MaxSectorAccessPrice))
+	}
+	if params.AcceptContracts {
+		if params.V2 {
+			filters = append(filters, "rhp4_settings_accepting_contracts = 1")
+		} else {
+			filters = append(filters, "settings_accepting_contracts = 1")
+		}
+	}
+	if params.Online {
+		filters = append(filters, "last_scan_successful = 1")
+	}
+	args = append(args, limit, offset)
+
+	var sortColumn string
+	switch sortBy {
+	case explorer.HostSortDateCreated:
+		sortColumn = "known_since"
+	case explorer.HostSortNetAddress:
+		sortColumn = "net_address"
+	case explorer.HostSortPublicKey:
+		sortColumn = "public_key"
+	case explorer.HostSortUptime:
+		sortColumn = uptimeValue
+	case explorer.HostSortAcceptingContracts:
+		if params.V2 {
+			sortColumn = "rhp4_settings_accepting_contracts"
+		} else {
+			sortColumn = "settings_accepting_contracts"
+		}
+	case explorer.HostSortStoragePrice:
+		if params.V2 {
+			sortColumn = "rhp4_prices_storage_price"
+		} else {
+			sortColumn = "settings_storage_price"
+		}
+	case explorer.HostSortContractPrice:
+		if params.V2 {
+			sortColumn = "rhp4_prices_contract_price"
+		} else {
+			sortColumn = "settings_contract_price"
+		}
+	case explorer.HostSortDownloadPrice:
+		if params.V2 {
+			sortColumn = "rhp4_prices_egress_price"
+		} else {
+			sortColumn = "settings_download_bandwidth_price"
+		}
+	case explorer.HostSortUploadPrice:
+		if params.V2 {
+			sortColumn = "rhp4_prices_ingress_price"
+		} else {
+			sortColumn = "settings_upload_bandwidth_price"
+		}
+	case explorer.HostSortUsedStorage:
+		if params.V2 {
+			sortColumn = "rhp4_settings_total_storage - rhp4_settings_remaining_storage"
+		} else {
+			sortColumn = "settings_total_storage - settings_remaining_storage"
+		}
+	case explorer.HostSortTotalStorage:
+		if params.V2 {
+			sortColumn = "rhp4_settings_total_storage"
+		} else {
+			sortColumn = "settings_total_storage"
+		}
+	default:
+		panic("invalid sort column")
```

Comment:
Naked panic? I haven't seen any validation in the header. This should eventually end up returning a 400 status code. I don't mind the store panicking that much but if we do that we should validate it higher up, ensuring that this panic is essentially a developer error.

Reply:
Honestly I'd just return an error here. No need to ever risk crashing the explorer over this.